tools/val/val.cpp

```diff
@@ -33,22 +33,31 @@ or if the filename is "-", then the binary is read from standard input.
 NOTE: The validator is a work in progress.
 
 Options:
-  -h, --help   Print this help.
-  --version    Display validator version information.
-  --target-env {vulkan1.0|spv1.0|spv1.1}
-               Use Vulkan1.0/SPIR-V1.0/SPIR-V1.1 validation rules.
+  -h, --help             Print this help.
+  --max_struct_members   <maximum number of structure members allowed>
+  --version              Display validator version information.
+  --target-env           {vulkan1.0|spv1.0|spv1.1}
+                         Use Vulkan1.0/SPIR-V1.0/SPIR-V1.1 validation rules.
 )",
       argv0, argv0);
 }
 
 int main(int argc, char** argv) {
   const char* inFile = nullptr;
+  const char* max_struct_members = nullptr;
   spv_target_env target_env = SPV_ENV_UNIVERSAL_1_1;
 
   for (int argi = 1; argi < argc; ++argi) {
     const char* cur_arg = argv[argi];
     if ('-' == cur_arg[0]) {
-      if (0 == strcmp(cur_arg, "--version")) {
+      if (0 == strcmp(cur_arg, "--max_struct_members")) {
```

Comment:
Or maybe that's too complex.  Shader compilation has about 50 limit settings. SPIR-V only about 15.